sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala

```diff
@@ -718,7 +718,17 @@ case class GpuLiteral (value: Any, dataType: DataType) extends GpuLeafExpression
   override def columnarEvalAny(batch: ColumnarBatch): Any = {
     // Returns a Scalar instead of the value to support the scalar of nested type, and
     // simplify the handling of result from a `expr.columnarEval`.
-    GpuScalar(value, dataType)
+    dataType match {
```

Comment:
This change here is useless.
Because we already have the type check and protection before moving it to GPU. Besides, this now is only for `Invoke`, so we can override the `childExprs` of `InvokeExprMeta` to exclude the targetObject and handle this `targetObject` literal internally and specially.
```
  override val childExprs: Seq[BaseExprMeta[_]] =
    invoke.arguments.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
```